pkg/repository/ghreleases/ghreleases.go

```diff
@@ -82,9 +82,54 @@ func (ghr *GHReleases) PublishProbe(driverVersion string, probePath string) erro
 }
 
 // IsAlreadyMirrored implmements repository.Repository.IsAlreadyMirrored for GitHub Releases.
-func (ghr *GHReleases) IsAlreadyMirrored(driverVersion string, probeName string) bool {
-	// TODO: unimplimented
-	return false
+func (ghr *GHReleases) IsAlreadyMirrored(driverVersion string, probeName string) (bool, error) {
+	// use a mutex to ensure that this function is only called once at a time as it is not thread safe.
+	// i.e. a release that doesn't exist may result in multiple goroutines trying to create it at once.
+	ghr.releasesMu.Lock()
+	defer ghr.releasesMu.Unlock()
+	ctx := context.Background()
+
+	// Retrieve the releases
+	currentReleasePage := 0
+	lastReleasePage := 0
```

Comment:
I suggest following the same pattern in their example: https://pkg.go.dev/github.com/google/go-github/v37/github#hdr-Pagination
e.g. for iterating through releases, it'd be
```go
        opt := &github.ListOptions{PerPage: 100}
	for {
		releases, releasesResp, err := ghr.ghClient.Repositories.ListReleases(ctx, ghr.owner, ghr.repo, opt)
		if err != nil {
			return false, fmt.Errorf("could not list releases: %w", err)
		}
		if resp.NextPage == 0 {
			break
		}
		opt.Page = resp.NextPage
	}
```

Reply:
I'm now using the example pattern, thanks!